android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt

```diff
@@ -406,6 +433,15 @@ class AddonManager(
         return Handler(iconThread.looper).asCoroutineDispatcher("WebExtensionIconDispatcher")
     }
 
+    private fun postSuccessfulAddonInstallation(ext: WebExtension, pendingAction: CompletableDeferred<Unit>): Addon {
```

Comment:
Maybe we could do this, and avoid returning an `addon` in addition to calling `registerXXX` and so on

```suggestion
    private fun onAddonInstalled(ext: WebExtension, pendingAction: CompletableDeferred<Unit>, onSuccess: ((Addon) -> Unit)) {
```

Reply:
oh, hm. and why is this not a problem for the newly introduced install method then?

Reply:
I think it's because we don't have the previous add-on to extract the `iconUrl`, I remembered we had a bug with the icon, I don't quite recall what was the issue about but we introduced the change [here](https://github.com/mozilla-mobile/firefox-android/commit/e73e3a7c4a9ea8de53e44618134000b3320a8721). My main worry is that we re introduce the issue 😅 .

Reply:
I guess it should be safe as we are loading the icon when we are installing

Reply:
I'll give a try without the `iconUrl` !

Reply:
I added the suggested changes!